tests/installation/iso_checksum.pm

```diff
@@ -0,0 +1,31 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify data integrity of the iso image by comparing checksum from IBS/OBS with checksum on the worker
+# Maintainer: Joaquín Rivera <jeriveramoya@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use testapi;
+use Digest::file 'digest_file_hex';
+
+sub run {
+    my $self = shift;
+
+    my $result   = 'fail';
+    my $iso      = get_required_var("ISO");
+    my $checksum = get_required_var("ISO_CHECKSUM");
+
+    my $digest = digest_file_hex($iso, "SHA-256");
+    diag "ISO checksum on the worker side : $digest";
+    $result = 'ok' if ($checksum eq $digest);
+    $self->result($result);
```

Comment:
The problem with `result('fail')` is that one has to go to the log to see what happened, nothing useful in the displaying system.

I suggest to use `record_info()` with `result => fail` with some reasonable text, when the check fails, and `ok` for the current `diag` text.

Reply:
Maybe just `die 'ISO checksum failed' unless ($checksum eq $digest);`?

Reply:
I would prefer to keep it informative and not die.

Reply:
As a matter of test module result, is `die` different to `$self->result('fail')`? `die` seems to set the test to fail state _and_ displaying the problem in the dashboard.